src/coreclr/jit/lsrabuild.cpp

```diff
@@ -2443,6 +2443,15 @@ void LinearScan::buildIntervals()
             // assert(block->isRunRarely());
         }
 
+#ifdef TARGET_X86
+        // On x86 w/ varargs CodeGen::genFnProlog trashs eax register which happens
+        // to be the hidden stub parameter, so mark it as killed here.
+        if ((block == compiler->fgFirstBB) && compiler->info.compIsVarArgs)
+        {
+            addKillForRegs(RBM_EAX, currentLoc + 1);
```

Comment:
Can you please switch this prolog code to use `REG_SCRATCH` and then unify this with the case above?

Reply:
Will do (likely tomorrow, going AFK soon)